ts/util/logger/rotatingPinoDest.ts

```diff
@@ -0,0 +1,106 @@
+// Copyright 2022 Signal Messenger, LLC
+// SPDX-License-Identifier: AGPL-3.0-only
+
+import fs from 'fs';
+import pino from 'pino';
+
+import { DURATION } from '../../session/constants';
+
+/**
+ * Keep at most rotated 3 files, so  4 files total including the "current" one
+ */
+const MAX_ROTATIONS = 3;
+
+/**
+ * Do the file rotation every 24 hours, that should be enough for multiple
+ * restarts and prevent the files getting too big
+ */
+const ROTATION_INTERVAL = 24 * DURATION.HOURS;
+
+const RETRY_DELAY = 12 * DURATION.SECONDS;
+
+// 5 seconds * 12 = 1 minute
+const MAX_RETRY_COUNT = 12;
+
+export type RotatingPinoDestOptionsType = Readonly<{
+  logFile: string;
+}>;
+
+export function createRotatingPinoDest({
+  logFile,
+}: RotatingPinoDestOptionsType): ReturnType<typeof pino.destination> {
+  const maxSavedLogFiles = MAX_ROTATIONS;
+
+  const boom = pino.destination({
+    dest: logFile,
+    sync: true,
+    mkdir: true,
+  });
+
+  let retryCount = 0;
+
+  const warn = (msg: string) => {
+    const line = JSON.stringify({
+      level: 40,
```

Comment:
Will need import LogLevel

```suggestion
      level: LogLevel.Warn,
```